src/js/lib/youtube.js

```diff
@@ -65,6 +69,50 @@ export function pimpYouTubePlayer(videoId, node, height, width, chapters) {
             });
         });
     });
+
+
+    function chapterTimer(youTubePlayer, playerTotalTime) {
+        let chapterCurrentProgress;
+        const playerCurrentTime = youTubePlayer.getCurrentTime();
+        const currentChapter = chapters.filter(function(value){
+            const chapStart = value.chapterTimestamp;
+            const chapNext = value.nextChapter || playerTotalTime;
+            if(playerCurrentTime >= chapStart && playerCurrentTime <= chapNext){
+                chapterCurrentProgress = (playerCurrentTime-chapStart)/(chapNext-chapStart);
+                return value;
+            }
+        });
+        if (currentChapter.length === 1){
+            const chapterAll = [].slice.call(document.querySelectorAll('li[data-sheet-timestamp]'));
+            chapterAll.forEach(function(el){
+                if (el.dataset.sheetTimestamp === currentChapter[0].chapterTimestamp){
+                    el.classList.add('docs--chapters-active');
+                    el.classList.remove('docs--chapters-inactive');
+                    const progress = el.querySelector('.progress');
+                    progress.style.width = `${chapterCurrentProgress*100}%`;
+                } else {
+                    el.classList.add('docs--chapters-inactive');
+                    el.classList.remove('docs--chapters-active');
+                }
+            });
+        }
+    }
+
+    function sendPercentageCompleteEvents(youTubePlayer, playerTotalTime) {
+        const quartile = playerTotalTime / 4;
+
+        let playbackEvents = new Map()
```

Comment:
hmm, a generator feels a little extreme to map over a simple structure imo. Might be simpler with a `Object.keys(playbackEvents).forEach`?

```js
Object.keys(playbackEvents).forEach(pbEvent => {
  console.log('the key is ' + key);  
  console.log('the value is ' + playbackEvents[key]);
});
```

Happy either way.